pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -599,10 +594,18 @@ func deployGardenlet(ctx context.Context, k8sGardenClient kubernetes.Interface,
 		componentImageVectorOverwrites = string(data)
 	}
 
+	gardenletImage, err := imageVector.FindImage("gardenlet")
+	if err != nil {
+		return err
+	}
 	var (
 		repository = gardenletImage.String()
-		tag        = version.Get().GitVersion
+		tag        string
 	)
+	tag, err = version.GetStableVersion(version.Get().GitVersion)
```

Comment:
As discussed, this will do a downgrade of deployed `gardenlet`s if the root `gardenlet` is running a `-dev` version and the last minor has patch releases, which were deployed bevor.